### PR TITLE
Added offset attribute

### PR DESCRIPTION
**Pull request by Vendin**
Added offset attribute for progress event

### CODE REVIEW

**Comment by sokil** [comment]
Thank you for contributing.

Due to specification 2.3.2.3 "Progress Event" https://www.iab.com/wp-content/uploads/2015/06/VASTv3_0.pdf attribute `offset` may be applied only for progress tracking, but you add offset to method that may add adny event.

I think you need something like:

```php
public function addProgressTrackingEvent($url, $offset) {

}
```

File: src/Creative/AbstractLinearCreative.php [hunk 1]
```diff
@@ -287,11 +288,49 @@ public function addTrackingEvent($event, $url)
         
         // add event attribute
         $trackingDomElement->setAttribute('event', $event);
-        
+
+        // add offset attribute
+        if (isset($offset)) {
```
Comment (sokil) [inline src/Creative/AbstractLinearCreative.php]: Due to specification 2.3.2.3 "Progress Event" https://www.iab.com/wp-content/uploads/2015/06/VASTv3_0.pdf attribute offset may be applied only for progress tracking, but you add offset to method that may add any event.

You may add specialied method for progress tracking and use default method as usually without offset

Also add to phpdoc that thi offset may be added as time value (HH:MM:SS or HH:MM:SS.mmm) or percentage (n%) value that indicates the timing for recording a view, or and integer value of seconds

**Review by sokil — CHANGES_REQUESTED** [review]

File: src/Creative/AbstractLinearCreative.php (hunk 1 above)
Comment (Vendin) [reply src/Creative/AbstractLinearCreative.php]: Thank you for feedback.

Yes you are right. The implementation with addProgressTrackingEvent is better. I will make changes.

File: src/Creative/AbstractLinearCreative.php [hunk 2]
```diff
@@ -287,11 +287,67 @@ public function addTrackingEvent($event, $url)
...
         $cdata = $this->linearCreativeDomElement->ownerDocument->createCDATASection($url);
         $trackingDomElement->appendChild($cdata);
-        
+
         return $this;
     }
+
+    /**
+     * @param string $url
+     * @param int|string $offset seconds or time in format "H:m:i"
```
Comment (sokil) [inline src/Creative/AbstractLinearCreative.php]: or percents in format "n%"

**Review by sokil — APPROVED** [review]

**Review by sokil — APPROVED** [review]

**Comment by sokil** [comment]
merged, thank you

**Pull request merged**